pyrogram/client/methods/messages/forward_messages.py

```diff
@@ -59,35 +69,53 @@ def forward_messages(self,
         Raises:
             :class:`Error <pyrogram.Error>` in case of a Telegram RPC error.
         """
+
         is_iterable = not isinstance(message_ids, int)
         message_ids = list(message_ids) if is_iterable else [message_ids]
 
-        r = self.send(
-            functions.messages.ForwardMessages(
-                to_peer=self.resolve_peer(chat_id),
-                from_peer=self.resolve_peer(from_chat_id),
-                id=message_ids,
-                silent=disable_notification or None,
-                random_id=[self.rnd_id() for _ in message_ids]
+        if as_copy:
+            sent_messages = []
+            for chunk in [message_ids[i:i+200] for i in range(0, len(message_ids), 200)]:
+                messages = self.get_messages(chat_id=from_chat_id, message_ids=chunk)  # type: pyrogram.Messages
+                for message in messages.messages:
+                    sent_messages.append(message.forward(
```

Comment:
Flood waits will break forwards in case of too many messages sent as-copies in a short period. We should handle such errors here.